code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstanceCreate.java

```diff
@@ -87,6 +92,30 @@ protected boolean shouldStart(Instance instance) {
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private List<Volume> processManagedVolumes(ProcessState state, Instance instance) {
+        List<Volume> volumes = new ArrayList<Volume>();
+        Map<String, Object> dataVolumeMounts = DataAccessor.field(instance, DATA_VOLUME_MOUNTS, Map.class);
```

Comment:
Use the `DataAccessor.fieldMap()` and `fieldString` and `fieldStringList`, etc.  Less casting madness